java-security/src/main/java/com/sap/cloud/security/token/validation/validators/JwtSignatureValidator.java

```diff
@@ -76,18 +77,20 @@ public ValidationResult validate(Token token) {
 		String jwksUri;
 		String keyId;
 		String appTidForTokenKeys = null;
+		String azpForTokenKeys = null;
 
 		if (Service.IAS == configuration.getService()) {
 			appTidForTokenKeys = token.getAppTid();
 			if (isTenantIdCheckEnabled && !token.getIssuer().equals("" + configuration.getUrl())
 					&& appTidForTokenKeys == null) {
 				return createInvalid("Error occurred during signature validation: OIDC token must provide app_tid.");
 			}
+			azpForTokenKeys = token.getClaimAsString(TokenClaims.AUTHORIZATION_PARTY);
 		}
 		try {
 			jwksUri = getOrRequestJwksUri(token);
 			String fallbackPublicKey = null;
-			if (configuration != null && configuration.hasProperty("verificationkey")) {
+			if (configuration.hasProperty("verificationkey")) {
```

Comment:
removed null check to solve warning because it cannot be null due to assertion in constructor